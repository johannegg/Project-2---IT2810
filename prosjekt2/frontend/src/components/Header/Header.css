.header {
	background-color: var(--theme-green);
	padding: 20px;
	display: flex;
	align-items: center;
	justify-content: flex-start;
	position: fixed;
	top: 0;
	left: 0;
	width: 100%;
	box-sizing: border-box;
	z-index: 1000;
	height: 80px;
}

.header-content {
	display: flex;
	align-items: center;
	max-width: 1200px;
	margin: 0 auto;
	position: relative;
	left: 10px;
}

.sofa-icon {
	height: 55px;
	width: 55px;
	margin-right: 15px;
}

.header-title {
	font-family: "Roboto", sans-serif;
	font-size: 30px;
	color: var(--theme-dark-grey);
	font-weight: bold;
	margin: 0;
}

.navbar {
	width: 30%;
	display: flex;
	align-items: center;
	justify-content: flex-end;
	flex-grow: 1;
}

.navButton {
	text-decoration: none;
	color: inherit;
	display: flex;
	align-items: center;
	white-space: nowrap;
	margin: 20px;
	background-color: transparent;
	border-bottom: solid 2px transparent;
}

.navButton:hover {
	text-decoration: none;
	color: inherit;
	border-bottom: solid 3px var(--theme-pink);
	padding-bottom: 7px;
	transition: all 0.3s ease;
}

.navIcon {
	margin: 10px;
}

.musicIcon {
	font-size: xx-large;
}

@media (max-width: 900px) {
	.header-title {
		font-size: 28px;
	}
	.navButton {
		font-size: medium;
		margin: 5px;
	}
	.sofa-icon {
		height: 50px;
		width: 50px;
	}
	.navIcon {
		font-size: 28px;
	}
}

@media (max-width: 750px) {
	.header-title {
		font-size: 24px;
	}
	.navButton {
		margin-left: 10px;
		margin-right: 5px;
	}

	.navText {
		display: none;
	}
	.navIcon {
		font-size: xx-large;
	}
}

@media (max-width: 600px) {
	.header-title {
		font-size: 22px;
	}
}

@media (max-width: 420px) {
	.header-title {
		font-size: 20px;
	}
	.header-width {
		width: 300px;
	}

	.navButton {
		margin: 2px;
	}

	.navIcon {
		font-size: 24px;
	}
}

@media (prefers-color-scheme: dark) {
	.header {
		background-color: var(--theme-even-darker-green);
	}

	.header-title {
		color: white;
	}

	.navButton:hover {
		border-color: var(--theme-dark-pink);
	}
}
