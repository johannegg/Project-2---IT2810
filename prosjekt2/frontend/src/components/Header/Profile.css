.profile {
    width: 180px;
    background-color: transparent;
    z-index: 10;
}

.profile-login {
    display: flex;
    flex-direction: row;
    align-items: center;
    background-color: white;
    border-radius: 5px;
    height: fit-content;
    padding-left: 2px;
    justify-content: space-between;
    right: 10px;
    min-width: 140px;
}

.login-input {
    border-color: transparent;
    width: 100%;
    outline: none;
}

.login-button {
    background-color: #E27396;
    border-top-right-radius: inherit;
    border-bottom-right-radius: inherit;
}
  
.dropdown {
    position: absolute;
    display: inline-block;
    right: 10px;
    bottom: 10px;
}

.dropdown-content {
    display: none;
    position: absolute;
    background-color: #f9f9f9;
    min-width: 140px;
    z-index: 1;
    right: 5px;
    cursor: pointer;
    border-radius: 5px;
  }
  
  .dropdown-content a {
    color: black;
    font-size: smaller;
    padding: 10px 10px;
    display: flex;
    flex-direction: column;
    border-radius: 5px;
  }
  
.dropdown-content a:hover {background-color: #f1f1f1}

.dropdown:hover .dropdown-content {
    display: block;
}

  