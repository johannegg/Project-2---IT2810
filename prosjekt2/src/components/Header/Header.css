.header {
    background-color: var(--theme-green); 
    padding: 20px; 
    display: flex; 
    align-items: center; 
    justify-content: flex-start; 
    position: fixed; 
    top: 0; 
    left: 0;
    width: 100%; 
    box-sizing: border-box; 
    z-index: 1000; 
    height: 70px;
  }
  
  .header-content {
    display: flex;
    align-items: center;
    justify-content: center; 
    width: 100%;
    max-width: 1200px; 
    margin: 0 auto; 
  }
  
  .sofa-icon {
    height: 60px;
    width: 60px;
    margin-right: 15px;
    filter: drop-shadow(2px 2px 3px rgba(0,0,0,0.15)); 
  }
  
  .header-title {
    font-family: 'Roboto', sans-serif;
    font-size: 28px; 
    color: #333; 
    font-weight: bold;
    margin: 0; 
  }
  