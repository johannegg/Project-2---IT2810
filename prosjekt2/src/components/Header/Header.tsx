import React from 'react';
import './Header.css'; 
import couchIcon from '../../assets/couch.svg'; 

const Header: React.FC = () => {
  return (
    <header className="header">
      <div className="header-content">
        <img
          src={couchIcon} 
          alt="Sofa Icon"
          className="sofa-icon"
        />
        <h1 className="header-title">Lyrical Lounge</h1>
      </div>
    </header>
  );
};

export default Header;
