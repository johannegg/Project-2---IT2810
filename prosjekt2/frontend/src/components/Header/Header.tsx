import "./Header.css";
import couchIcon from "../../assets/couch.svg";
import { Link, useNavigate } from "react-router-dom";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
import { faHeart } from "@fortawesome/free-regular-svg-icons";
import { BsMusicNoteList } from "react-icons/bs";
import Profile from "../Profile/Profile";

const Header: React.FC = () => {
	const navigate = useNavigate();
	const handleNavigation = (path: string, event: React.MouseEvent<HTMLAnchorElement, MouseEvent>) => {
		const username = localStorage.getItem("profileName");
		if (username && username !== "") {
			navigate(path);
		} else {
			event.preventDefault();
			alert("You need to log in to access this page.");
		}
	};

	return (
		<header className="header">
			<Link to="/" className="header-content">
				<img src={couchIcon} alt="Sofa Icon" className="sofa-icon" />
				<h1 className="header-title">Lyrical Lounge</h1>
			</Link>
			<nav className="navbar">
				<Link
					to="/favorites"
					onClick={(e) => handleNavigation("/favorites", e)}
					className={`navButton ${location.pathname === "/favorites" ? "active" : ""}`}
				>
					<FontAwesomeIcon icon={faHeart} size="2x" className="navIcon" />
					<span className="navText">Favorited songs</span>
				</Link>
				<Link
					to="/playlists"
					onClick={(e) => handleNavigation("/playlists", e)}
					className={`navButton ${location.pathname === "/playlists" ? "active" : ""}`}
				>
					<BsMusicNoteList className="navIcon musicIcon" />
					<span className="navText">Your playlists</span>
				</Link>
				<Profile />
			</nav>
		</header>
	);
};

export default Header;
