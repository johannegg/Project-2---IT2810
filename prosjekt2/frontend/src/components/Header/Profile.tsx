import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faArrowRight, faCircleUser } from '@fortawesome/free-solid-svg-icons';
import { useEffect, useState } from 'react';
import "./Profile.css";

const Profile: React.FC = () => {
    const [profileName, setProfile] = useState<string>("");
    const [isLoggedIn, setLogin] = useState<boolean>(false);
    const [inputValue, setInputValue] = useState<string>("");
    const [showLogin, setShowLogin] = useState<boolean>(false);

    const logOut = () => {
        setLogin(false);
        localStorage.removeItem("profileName");
        setShowLogin(false)
    };

    const logIn = () => {
        if (inputValue.trim() !== "") {
            setProfile(inputValue);
            setLogin(true);
            localStorage.setItem("profileName", inputValue);
        }
    };
    
    useEffect(() => {
        const storedProfileName = localStorage.getItem("profileName") || "";
        if (storedProfileName) {
            setProfile(storedProfileName);
            setLogin(true);
        }
    }, []);

    
    return (
        <>
            {isLoggedIn ? (
                <div className="dropdown">
                    <button className="profile-icon">
                        <FontAwesomeIcon icon={faCircleUser} size="2xl"/>
                    </button>
                    <div className="dropdown-content">
                        <a onClick={logOut}>
                            Log out
                        </a>
                    </div>
                </div>
            ) : (
                <>
                    <div className="dropdown">
                        <button className="profile-icon" onClick={() => setShowLogin(!showLogin)}>
                            <FontAwesomeIcon icon={faCircleUser} size="2xl"/>
                        </button>
                    </div>
                    {showLogin &&
                        <div className="login-overlay">
                        <form className="profile-login" onSubmit={(e) => { e.preventDefault(); logIn(); }}>  
                            <FontAwesomeIcon icon={faCircleUser} size="lg" style={{ color: "#E27396" }}/>
                            <input
                                className="login-input"
                                placeholder="Enter profile name"
                                onChange={(e) => setInputValue(e.target.value)}
                                />
                            <button className="login-button" type="submit">
                                <FontAwesomeIcon icon={faArrowRight} style={{ color: "#FFF" }}/>
                            </button>
                        </form>
                    </div>
                    }
                </>
            )}
        </>
    );
};

export default Profile;
