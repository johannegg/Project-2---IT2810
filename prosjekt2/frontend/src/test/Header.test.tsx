import { describe, test, vi, beforeEach, expect } from "vitest";
import { render, screen, fireEvent } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";
import Header from "../components/Header/Header";

// Correctly mock useNavigate
const navigateMock = vi.fn();
vi.mock("react-router-dom", async () => {
    const actual = await vi.importActual("react-router-dom");
    return {
        ...actual,
        useNavigate: () => navigateMock,
    };
});

// Mock the Profile component
vi.mock("../components/Profile/Profile", () => ({
    default: () => <div>Mock Profile</div>,
}));

describe("Header Component", () => {
    beforeEach(() => {
        // Reset mocks and localStorage before each test
        localStorage.clear();
        navigateMock.mockReset();
    });

    test("renders the header component", () => {
        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );

        // Check for header elements
        expect(screen.getByAltText("Sofa Icon")).toBeInTheDocument();
        expect(screen.getByText(/lyrical lounge/i)).toBeInTheDocument();
        expect(screen.getByText(/favorited songs/i)).toBeInTheDocument();
        expect(screen.getByText(/your playlists/i)).toBeInTheDocument();
        expect(screen.getByText(/mock profile/i)).toBeInTheDocument();

        // Take a snapshot of the rendered Header
        expect(container).toMatchSnapshot();
    });

    test("navigates to homepage when the title is clicked", () => {
        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );

        const titleLink = screen.getByText(/lyrical lounge/i).closest("a");
        expect(titleLink).toHaveAttribute("href", "/");

        // Take a snapshot of the navigation behavior
        expect(container).toMatchSnapshot();
    });

    test("alerts and prevents navigation if not logged in", () => {
        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );

        const favoritesLink = screen.getByText(/favorited songs/i).closest("a");
        const playlistsLink = screen.getByText(/your playlists/i).closest("a");

        // Mock the alert function
        window.alert = vi.fn();

        // Simulate clicking on links without being logged in
        fireEvent.click(favoritesLink!);
        fireEvent.click(playlistsLink!);

        expect(window.alert).toHaveBeenCalledWith("You need to log in to access this page.");
        expect(navigateMock).not.toHaveBeenCalled();

        // Take a snapshot of the behavior
        expect(container).toMatchSnapshot();
    });

    test("navigates to favorites or playlists if logged in", () => {
        localStorage.setItem("profileName", "testUser");

        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );

        const favoritesLink = screen.getByText(/favorited songs/i).closest("a");
        const playlistsLink = screen.getByText(/your playlists/i).closest("a");

        // Simulate clicking on links when logged in
        fireEvent.click(favoritesLink!);
        fireEvent.click(playlistsLink!);

        expect(navigateMock).toHaveBeenCalledWith("/favorites");
        expect(navigateMock).toHaveBeenCalledWith("/playlists");

        // Take a snapshot of the logged-in navigation behavior
        expect(container).toMatchSnapshot();
    });

    test("highlights the active navigation link", () => {
        // Mock location.pathname
        Object.defineProperty(window, "location", {
            value: { pathname: "/favorites" },
            writable: true,
        });

        const { container } = render(
            <MemoryRouter>
                <Header />
            </MemoryRouter>
        );

        const favoritesLink = screen.getByText(/favorited songs/i).closest("a");
        const playlistsLink = screen.getByText(/your playlists/i).closest("a");

        expect(favoritesLink).toHaveClass("active");
        expect(playlistsLink).not.toHaveClass("active");

        // Take a snapshot of the active link highlighting
        expect(container).toMatchSnapshot();
    });
});
