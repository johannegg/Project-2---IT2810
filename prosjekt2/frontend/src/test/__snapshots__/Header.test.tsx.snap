// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > alerts and prevents navigation if not logged in 1`] = `
<div>
  <header
    aria-label="Main header"
    class="header"
  >
    <a
      aria-label="Go to home page"
      class="header-content"
      href="/"
    >
      <img
        alt="Sofa Icon"
        class="sofa-icon"
        src="/src/assets/couch.svg"
      />
      <h1
        class="header-title"
      >
        Lyrical Lounge
      </h1>
    </a>
    <nav
      aria-label="Main navigation"
      class="navbar"
    >
      <a
        aria-label="Go to favorites page"
        class="navButton "
        href="/favorites"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-2x navIcon"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
        <span
          class="navText"
        >
          Favorited songs
        </span>
      </a>
      <a
        aria-label="Go to playlists page"
        class="navButton "
        href="/playlists"
      >
        <svg
          class="navIcon musicIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2"
          />
          <path
            d="M12 3v10h-1V3z"
            fill-rule="evenodd"
          />
          <path
            d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z"
          />
          <path
            d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="navText"
        >
          Your playlists
        </span>
      </a>
      <div>
        Mock Profile
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component > highlights the active navigation link 1`] = `
<div>
  <header
    aria-label="Main header"
    class="header"
  >
    <a
      aria-label="Go to home page"
      class="header-content"
      href="/"
    >
      <img
        alt="Sofa Icon"
        class="sofa-icon"
        src="/src/assets/couch.svg"
      />
      <h1
        class="header-title"
      >
        Lyrical Lounge
      </h1>
    </a>
    <nav
      aria-label="Main navigation"
      class="navbar"
    >
      <a
        aria-label="Go to favorites page"
        class="navButton active"
        href="/favorites"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-2x navIcon"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
        <span
          class="navText"
        >
          Favorited songs
        </span>
      </a>
      <a
        aria-label="Go to playlists page"
        class="navButton "
        href="/playlists"
      >
        <svg
          class="navIcon musicIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2"
          />
          <path
            d="M12 3v10h-1V3z"
            fill-rule="evenodd"
          />
          <path
            d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z"
          />
          <path
            d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="navText"
        >
          Your playlists
        </span>
      </a>
      <div>
        Mock Profile
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component > navigates to favorites or playlists if logged in 1`] = `
<div>
  <header
    aria-label="Main header"
    class="header"
  >
    <a
      aria-label="Go to home page"
      class="header-content"
      href="/"
    >
      <img
        alt="Sofa Icon"
        class="sofa-icon"
        src="/src/assets/couch.svg"
      />
      <h1
        class="header-title"
      >
        Lyrical Lounge
      </h1>
    </a>
    <nav
      aria-label="Main navigation"
      class="navbar"
    >
      <a
        aria-label="Go to favorites page"
        class="navButton "
        href="/favorites"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-2x navIcon"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
        <span
          class="navText"
        >
          Favorited songs
        </span>
      </a>
      <a
        aria-label="Go to playlists page"
        class="navButton "
        href="/playlists"
      >
        <svg
          class="navIcon musicIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2"
          />
          <path
            d="M12 3v10h-1V3z"
            fill-rule="evenodd"
          />
          <path
            d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z"
          />
          <path
            d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="navText"
        >
          Your playlists
        </span>
      </a>
      <div>
        Mock Profile
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component > navigates to homepage when the title is clicked 1`] = `
<div>
  <header
    aria-label="Main header"
    class="header"
  >
    <a
      aria-label="Go to home page"
      class="header-content"
      href="/"
    >
      <img
        alt="Sofa Icon"
        class="sofa-icon"
        src="/src/assets/couch.svg"
      />
      <h1
        class="header-title"
      >
        Lyrical Lounge
      </h1>
    </a>
    <nav
      aria-label="Main navigation"
      class="navbar"
    >
      <a
        aria-label="Go to favorites page"
        class="navButton "
        href="/favorites"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-2x navIcon"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
        <span
          class="navText"
        >
          Favorited songs
        </span>
      </a>
      <a
        aria-label="Go to playlists page"
        class="navButton "
        href="/playlists"
      >
        <svg
          class="navIcon musicIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2"
          />
          <path
            d="M12 3v10h-1V3z"
            fill-rule="evenodd"
          />
          <path
            d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z"
          />
          <path
            d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="navText"
        >
          Your playlists
        </span>
      </a>
      <div>
        Mock Profile
      </div>
    </nav>
  </header>
</div>
`;

exports[`Header Component > renders the header component 1`] = `
<div>
  <header
    aria-label="Main header"
    class="header"
  >
    <a
      aria-label="Go to home page"
      class="header-content"
      href="/"
    >
      <img
        alt="Sofa Icon"
        class="sofa-icon"
        src="/src/assets/couch.svg"
      />
      <h1
        class="header-title"
      >
        Lyrical Lounge
      </h1>
    </a>
    <nav
      aria-label="Main navigation"
      class="navbar"
    >
      <a
        aria-label="Go to favorites page"
        class="navButton "
        href="/favorites"
      >
        <svg
          aria-hidden="true"
          class="svg-inline--fa fa-heart fa-2x navIcon"
          data-icon="heart"
          data-prefix="far"
          focusable="false"
          role="img"
          viewBox="0 0 512 512"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M225.8 468.2l-2.5-2.3L48.1 303.2C17.4 274.7 0 234.7 0 192.8l0-3.3c0-70.4 50-130.8 119.2-144C158.6 37.9 198.9 47 231 69.6c9 6.4 17.4 13.8 25 22.3c4.2-4.8 8.7-9.2 13.5-13.3c3.7-3.2 7.5-6.2 11.5-9c0 0 0 0 0 0C313.1 47 353.4 37.9 392.8 45.4C462 58.6 512 119.1 512 189.5l0 3.3c0 41.9-17.4 81.9-48.1 110.4L288.7 465.9l-2.5 2.3c-8.2 7.6-19 11.9-30.2 11.9s-22-4.2-30.2-11.9zM239.1 145c-.4-.3-.7-.7-1-1.1l-17.8-20-.1-.1s0 0 0 0c-23.1-25.9-58-37.7-92-31.2C81.6 101.5 48 142.1 48 189.5l0 3.3c0 28.5 11.9 55.8 32.8 75.2L256 430.7 431.2 268c20.9-19.4 32.8-46.7 32.8-75.2l0-3.3c0-47.3-33.6-88-80.1-96.9c-34-6.5-69 5.4-92 31.2c0 0 0 0-.1 .1s0 0-.1 .1l-17.8 20c-.3 .4-.7 .7-1 1.1c-4.5 4.5-10.6 7-16.9 7s-12.4-2.5-16.9-7z"
            fill="currentColor"
          />
        </svg>
        <span
          class="navText"
        >
          Favorited songs
        </span>
      </a>
      <a
        aria-label="Go to playlists page"
        class="navButton "
        href="/playlists"
      >
        <svg
          class="navIcon musicIcon"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 16 16"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 13c0 1.105-1.12 2-2.5 2S7 14.105 7 13s1.12-2 2.5-2 2.5.895 2.5 2"
          />
          <path
            d="M12 3v10h-1V3z"
            fill-rule="evenodd"
          />
          <path
            d="M11 2.82a1 1 0 0 1 .804-.98l3-.6A1 1 0 0 1 16 2.22V4l-5 1z"
          />
          <path
            d="M0 11.5a.5.5 0 0 1 .5-.5H4a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 7H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5m0-4A.5.5 0 0 1 .5 3H8a.5.5 0 0 1 0 1H.5a.5.5 0 0 1-.5-.5"
            fill-rule="evenodd"
          />
        </svg>
        <span
          class="navText"
        >
          Your playlists
        </span>
      </a>
      <div>
        Mock Profile
      </div>
    </nav>
  </header>
</div>
`;
